bin/dfx-snapshot-mk-stock

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+set -euo pipefail
+SOURCE_DIR="$(dirname "$(realpath "${BASH_SOURCE[0]}")")"
+PATH="$SOURCE_DIR:$PATH"
+
+print_help() {
+  cat <<-EOF
+
+	Creates a snapshot of a stock environment, for use in CI tests.
+	EOF
+}
+
+# Source the clap.bash file ---------------------------------------------------
+source "$SOURCE_DIR/clap.bash"
+# Define options
+clap.define short=s long=snapshot desc="The file to save to" variable=DFX_SNAPSHOT_FILE default="stock-snsdemo-snapshot.tar.xz"
+# Source the output file ----------------------------------------------------------
+source "$(clap.build)"
+
+: Set up SNS state and create one finalized SNS
+bin/dfx-sns-demo
+
+: Stop the replica to let it persist all its state before we save the state.
```

Comment:
Note: This logic is probably always wanted before dfx-snapshot-save so can be moved there.